scripts/build.js

```diff
@@ -89,13 +93,88 @@ function checkChecksum(url, local_file) {
+ * so it doesn't leave a zombie package in your node_modules. To remove the package, call npmDeleteRuntimePackage
+ *
+ * @param {*} package_name The name of the package you want to download (example: 'cmake-js')
+ * @param {*} package_version The version of the package to download - leave blank for latest. (example: '6.3.2')
+ * @returns True if the package was downloaded dynamically, otherwise false.
+ */
+function npmDownloadAndInstallRuntimePackage(package_name, package_version=null) {
+    console.log("Looking for " + package_name + " as a dependency...");
+
+    // Do we have it in node modules? If so, then use that!
+    try {
+        if (fs.existsSync("./node_modules/" + package_name)) {
+            console.log("Found " + package_name + " in node_modules!");
+            return false;
+        }
+    } catch (error) {}
+
+    // Do we have it in our node list? If so, then use that!
+    try {
+        var list_output = child_process.execSync("npm list " + package_name, {encoding: "utf8"});
+        if (list_output.indexOf(package_name) !== -1) {
+            console.log("Found " + package_name + " in npm list!");
+            return false;
+        }
+    } catch (error) {}
+
+    // If it is not found in either, then download it into our node_modules
+    try {
+        console.log("Could not find " + package_name);
+        console.log("Downloading " + package_name + " from npm for build...");
+        // Try to intall the given package and ONLY the given package. Will throw an exception if there is an error.
+        if (package_version != null) {
+            child_process.execSync("npm install --no-package-lock --save-dev --ignore-scripts " + package_name + "@" + package_version);
+        } else {
+            child_process.execSync("npm install --no-package-lock --save-dev --ignore-scripts " + package_name);
+        }
+        return true;
+
+    } catch (err) {
+        console.log("ERROR - npm could not download " + package_name + "! " + package_name + " is required to build the CRT");
+        throw err;
+    }
+}
+
+/**
+ * Tells NPM to uninstall a package. This should only be used to clean up a dynamic package downloaded with the
+ * npmDownloadAndInstallRuntimePackage function, as otherwise it could remove a non-dynamic package.
+ * @param {*} package_name The name of the package you want to delete (example 'cmake-js')
+ */
+function npmDeleteRuntimePackage(package_name) {
+    console.log("Removing " + package_name + "...");
+    try {
+        child_process.execSync("npm uninstall " + package_name);
+    } catch (err) {
+        console.log("ERROR - npm could not remove " + package_name + "!");
+        throw err;
+    }
+}
+
+
 async function fetchNativeCode(url, version, path) {
+    // Get tar if it doesn't exist
+    var remove_tar_at_end = npmDownloadAndInstallRuntimePackage("tar", tar_version);
+    var tar = require('tar');
```

Comment:
Would we need a try-catch here to make sure the download succeed? While, I'm not sure how would js handle the errors, ignore it if it is not necessary here.

Reply:
I'm not sure right off. I think it would just stop the script from running, which would in turn prevent the build. However, I should test to find out 👍

Reply:
Just checked and it crashes, which is probably fine, but it generates a lot of error messages and stuff that is a little cluttered and hides the real cause of the download failing. Catching the error and exiting with an error code of 1 looks much better and is easier to read.

I adjusted the PR accordingly.